tests/framework/extensions/workloads/deployments/list.go

```diff
@@ -0,0 +1,63 @@
+package deployments
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/api/scheme"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads"
+	"github.com/sirupsen/logrus"
+	appv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// DeploymentList is a struct that contains a list of deployments.
+type DeploymentList struct {
+	Items []*appv1.Deployment
+}
+
+// ListDeployments is a helper function that uses the steve client to list deployments on a namespace for a specific cluster with its list options.
+func ListDeployments(client *rancher.Client, clusterID string, opts *types.ListOpts) (*DeploymentList, error) {
```

Comment:
Same as above I think this helper is fine.

Reply:
I think that would mean having exact same helpers with `kubeapi` extensions while this could've been done with one or two more additional lines in the tests. What are your thoughts?

Reply:
Are we going to use kubeapi extensions, since we're using steve client instead of dynamic client now?

Reply:
Yes, we want to give test users that option, if they want to test the Kubernetes API without the rancher layer, more likely in a integration test. Does that make sense?

Reply:
Yeah right, so keeping this helper function for the steve api, as there's a specific conversion.